test/brsTypes/components/RoSGNode.test.js

```diff
@@ -986,6 +986,741 @@ describe("RoSGNode", () => {
+                expect(result.elements[0]).toEqual(child1);
+                expect(result.elements[2]).toEqual(child2);
+            });
+            it("replaces existing child amongst 4 children at the end", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChildren = parent.getMethod("getchildren");
+                let replaceChild = parent.getMethod("replacechild");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+                appendChild.call(interpreter, child3);
+                appendChild.call(interpreter, child4);
+
+                let childReplaced = replaceChild.call(interpreter, child2, new Int32(3));
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+
+                expect(childReplaced.value).toEqual(true);
+                expect(result.elements.length).toEqual(4);
+                expect(result.elements[0]).toEqual(child1);
+                expect(result.elements[1]).toEqual(child3);
+                expect(result.elements[3]).toEqual(child2);
+            });
+        });
+        describe("replacechildren", () => {
+            it("replaces children with new nodes", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChildren = parent.getMethod("getchildren");
+                let replaceChildren = parent.getMethod("replacechildren");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+
+                let childReplaced = replaceChildren.call(
+                    interpreter,
+                    new RoArray([child3, child4]),
+                    new Int32(0)
+                );
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                expect(childReplaced.value).toEqual(true);
+                expect(result.elements.length).toEqual(2);
+                expect(result.elements[0]).toEqual(child3);
+            });
+            it("replaces children with new nodes that are more than number of children", () => {
+                let appendChild = parent.getMethod("appendchild");
+                let getChildren = parent.getMethod("getchildren");
+                let replaceChildren = parent.getMethod("replacechildren");
+
+                appendChild.call(interpreter, child1);
+
+                let childReplaced = replaceChildren.call(
+                    interpreter,
+                    new RoArray([child2, child3]),
+                    new Int32(0)
+                );
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                expect(childReplaced.value).toEqual(true);
+                expect(result.elements.length).toEqual(1);
+                expect(result.elements[0]).toEqual(child2);
+            });
```

Comment:
nit: I think we usually have a line of whitespace after closing an `it` statement